### PR TITLE
provide etcd public lb support option

### PR DESCRIPTION
**Pull request by srirg**

### CODE REVIEW

**Comment by srirg** [comment]
This change is for this issue.  https://github.com/oracle/terraform-kubernetes-installer/issues/73

**Comment by jlamillan** [comment]
Sri, be sure to pull in the most recent changes. We noticed some buggy code around the yum installation of the kubelet that was causing all kinds of weird problems. With the most recent changes, a vanilla deployment of a private cluster was functioning _properly_.

**Comment by srirg** [comment]
Jesse, I am still seeing NotReady issues if I run it multiples times for a vanilla private cluster.

File: network/vcn/securitylists.tf [hunk 1]
```diff
@@ -225,5 +225,14 @@ resource "oci_core_security_list" "PublicSecurityList" {
         "code" = 4
       }
     },
+    {
+      protocol = "6"
+      source   = "${var.public_subnet_ssh_ingress}"
```
Comment (jlamillan) [inline network/vcn/securitylists.tf]: This should be `var.etcd_cluster_ingress`.

File: network/loadbalancers/etcd/variables.tf [hunk 2]
```diff
@@ -34,3 +36,7 @@ variable "etcd_ad3_private_ips" {
 variable "shape" {
   default = "100Mbs"
 }
+
+variable "is_private" {
+  default = "false"
```
Comment (jlamillan) [inline network/loadbalancers/etcd/variables.tf]: The default for this should be private.

File: variables.tf [hunk 3]
```diff
@@ -302,6 +302,11 @@ variable "k8s_master_lb_access" {
   default     = "public"
 }
 
+variable "etcd_lb_access" {
+  description = "Whether etcd load balancer is launched in a public or private subnet"
+  default     = "public"
```
Comment (jlamillan) [inline variables.tf]: The default should be private here.

File: docs/input-variables.md [hunk 4]
```diff
@@ -21,6 +21,7 @@ name                                | default     | description
 ------------------------------------|-------------|------------
 control_plane_subnet_access         | public      | Whether instances in the control plane are launched in a public or private subnets
 k8s_master_lb_access                | public      | Whether the Kubernetes Master Load Balancer is launched in a public or private subnets
+etcd_lb_access                	    | public      | Whether the etcd Load Balancer is launched in a public or private subnets
```
Comment (jlamillan) [inline docs/input-variables.md]: The default value of etcd_lb_access should be private.

**Pull request merged**